sydent/validators/common.py

```diff
@@ -15,18 +21,32 @@ def validateSessionWithToken(sydent, sid, clientSecret, token):
     Attempt to validate a session, identified by the sid, using
     the token from out-of-band. The client secret is given to
     prevent attempts to guess the token for a sid.
-    If the session was sucessfully validated, return a dict
-    with 'success': True that can be sent to the client,
-    otherwise return False.
+
+    :param sid: The ID of the session to validate.
+    :type sid: unicode
+    :param clientSecret: The client secret to validate.
+    :type clientSecret: unicode
+    :param token: The token to validate.
+    :type token: unicode
+
+    :return: A dict with a "success" key which is True if the session
+        was successfully validated, False otherwise.
+    :rtype: dict[str, bool]
+
+    :raise IncorrectClientSecretException: The provided client_secret is incorrect.
+    :raise SessionExpiredException: The session has expired.
+    :raise InvalidSessionIdException: The session ID couldn't be matched with an
+        existing session.
+    :raise IncorrectSessionTokenException: The provided token is incorrect
     """
     valSessionStore = ThreePidValSessionStore(sydent)
     s = valSessionStore.getTokenSessionById(sid)
     if not s:
-        logger.info("Session ID %s not found", (sid))
-        return False
+        logger.info("Session ID %s not found", (sid,))
+        raise InvalidSessionIdException()
```

Comment:
This doesn't looke python3 related? Should this be in a separate PR?

Reply:
It is in a way, since it's making the return type of the function consistent thus easier to deal with.